scripts/migrateHistoricBids.ts

```diff
@@ -40,7 +40,7 @@ const main = async () => {
   const bidsPlacedMessages = bidsPlaced.map((x) => mapBidtoBidPlacedMessage(x));
   const bidsPlacedArchivedMessages = bidsCancelled.map((x) => mapBidtoBidPlacedMessage(x));
   const bidsCancelledMessages = bidsCancelled.map((x) =>
-    mapBidtoBidCancelledMessage(x)
+    mapBidtoBidCancelledMessage(x as CanceledBid)
```

Comment:
Seems a bit off. Is `x` a `Bid` or a `CanceledBid`?

Reply:
X is a Bid, an CanceledBid is a child of Bid

Is an intersection type necessary if canceledBid would already have all the properties of Bid because of its inheritance?

Reply:
It's not just an intersection.

I want you to make a type that is the proper combination of `Bid` and `CanceledBid`. You can't convert a bid -> CanceledBid because CanceledBid has an extra member. Create a type that properly expresses "Bid OR CanceledBid" that would fit this use case.

Reply:
I think it might be simpler just to get them from the database as CanceledBids

Reply:
Are they always going to be `CanceledBids` in the database?

Reply:
Mongo doesnt really have a schema, so it could be anything at any time lol

Ill try to implement the Bid OR CancelledBid type like you suggested